Userland/Libraries/LibJS/Runtime/ExecutionContext.cpp

```diff
@@ -13,9 +13,34 @@
 
 namespace JS {
 
+class ExecutionContextAllocator {
+public:
+    NonnullOwnPtr<ExecutionContext> allocate(Heap& heap)
+    {
+        if (m_execution_contexts.is_empty())
+            return adopt_own(*new ExecutionContext(heap));
+        void* slot = m_execution_contexts.take_last();
+        return adopt_own(*new (slot) ExecutionContext(heap));
+    }
+    void deallocate(ExecutionContext* ec)
```

Comment:
If this takes a `void*` then you don't need to cast in `operator delete` anymore :)

Reply:
Oh, and `ec` should be something like `context` as we don't generally abbreviate randomly like this.